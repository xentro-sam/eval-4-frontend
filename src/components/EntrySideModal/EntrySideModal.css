#entry-side-modal {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background-color: rgba(0, 0, 0, 0.5);
    display: flex;
    justify-content: center;
    align-items: flex-end;
    flex-direction: column;
    font: 1vw;
}

#side-modal-content {
    background-color: white;
    width: 45vw;
    height: 100vh;
    border-radius: 15px;
    display: flex;
    justify-content: center;
    flex-direction: column;
    padding: 0.25vw 3vw;
}

#side-modal-title {
    font-size: 1.5vw;
    font-weight: 600;
    margin-bottom: 1vw;
}

#side-modal-input > input {
    width: 100%;
    height: 2vw;
    border: 1px solid rgb(200, 210, 250);
    border-radius: 4px;
    box-sizing: border-box;
    font-size: 1vw;
    padding: 0.25vw;

}

#side-modal-buttons {
    display: flex;
    justify-content: flex-end;
    margin-top: 1vw;
}

#side-modal-cancel {
    width: 5vw;
    padding-top: 0.5vw;
}

#side-modal-create {
    width: 10vw;
    height: 2.5vw;
    margin-left: 1vw;
    border-radius: 5px;
    background-color: rgb(122, 97, 246);
    color: white;
}