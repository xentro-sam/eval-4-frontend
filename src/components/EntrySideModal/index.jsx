import * as React from 'react';
import './EntrySideModal.css';
import PropTypes from 'prop-types';
import makeRequest from '../../utils/makeRequest';

export default function EntrySideModal(props) {
  if (!props.show) {
    return null;
  }

  const handleCreate = async () => {
    const inputs = document.querySelectorAll('#side-modal-input input');
    const data = {};
    inputs.forEach((input) => {
      if (input.value === '') {
        return;
      }
      data[input.parentNode.querySelector('#side-modal-input-title').innerHTML] = input.value;
    });
    if (props.task === 'New') {
      makeRequest(props.api(props.contentTypeId), {
        data,
      }).then((response) => {
        props.createEntry(response);
      });
    } else {
      makeRequest(props.api(props.contentTypeId, props.postId), {
        data,
      }).then((response) => {
        props.updateEntry(response);
      });
    }
  };

  return (
    <div id="entry-side-modal">
      <div id="side-modal-content">
        <div id="side-modal-title">
          {props.task} {props.contentType}
        </div>
        <div id="side-modal-inputs">
          {props.fields.map((field) => {
            return field !== 'id' && (
              <div id="side-modal-input" key={field}>
                <div id="side-modal-input-title">{field}</div>
                <input type="text" />
              </div>
            );
          })}
        </div>
        <div id="side-modal-buttons">
          <div id="side-modal-cancel" onClick={props.onClose}>Cancel</div>
          <button type="submit" id="side-modal-create" onClick={handleCreate}>Add</button>
        </div>
      </div>
    </div>
  );
}

EntrySideModal.propTypes = {
  show: PropTypes.bool,
  contentType: PropTypes.string.isRequired,
  contentTypeId: PropTypes.number.isRequired,
  fields: PropTypes.array.isRequired,
  onClose: PropTypes.func.isRequired,
  createEntry: PropTypes.func,
  updateEntry: PropTypes.func,
  api: PropTypes.func.isRequired,
  task: PropTypes.string.isRequired,
  postId: PropTypes.number,
};
