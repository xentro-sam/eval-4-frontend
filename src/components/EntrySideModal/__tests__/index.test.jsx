import React from 'react';
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import EntrySideModal from '../index';
import {waitFor} from '@testing-library/dom';
import makeRequest from '../../../utils/makeRequest';
import {act} from 'react-dom/test-utils';

jest.mock('../../../utils/makeRequest');

const mockData = {
  show: true,
  contentType: 'mock title 1',
  contentTypeId: 1,
  fields: ['name', 'description'],
  onClose: jest.fn(),
  createEntry: jest.fn(),
  updateEntry: jest.fn(),
  api: jest.fn(),
  task: 'New',
  postId: 1,
};

describe('ContentType', () => {
  it('should render correctly', async () => {
    expect(makeRequest).not.toHaveBeenCalled();
    const {asFragment} = await act( async () => render(<EntrySideModal {...mockData} />));
    await waitFor(() => {
      expect(screen.getByText('Add')).toBeTruthy();
    });
    expect(asFragment()).toMatchSnapshot();
  });
});
