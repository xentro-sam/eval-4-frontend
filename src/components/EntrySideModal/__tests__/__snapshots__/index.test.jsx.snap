// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentType should render correctly 1`] = `
<DocumentFragment>
  <div
    id="entry-side-modal"
  >
    <div
      id="side-modal-content"
    >
      <div
        id="side-modal-title"
      >
        New mock title 1
      </div>
      <div
        id="side-modal-inputs"
      >
        <div
          id="side-modal-input"
        >
          <div
            id="side-modal-input-title"
          >
            name
          </div>
          <input
            type="text"
          />
        </div>
        <div
          id="side-modal-input"
        >
          <div
            id="side-modal-input-title"
          >
            description
          </div>
          <input
            type="text"
          />
        </div>
      </div>
      <div
        id="side-modal-buttons"
      >
        <div
          id="side-modal-cancel"
        >
          Cancel
        </div>
        <button
          id="side-modal-create"
          type="submit"
        >
          Add
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
